java/compaction/compaction-status-store/src/main/java/sleeper/compaction/status/store/task/DynamoDBCompactionTaskStatusFormat.java

```diff
@@ -110,15 +110,15 @@ private static void addStatusUpdate(Map<String, AttributeValue> item, Compaction
                         getInstantAttribute(item, EXPIRY_DATE, Instant::ofEpochSecond));
                 break;
             case FINISHED:
-                builder.taskFinished(taskId, CompactionTaskFinishedStatus.builder()
+                CompactionTaskFinishedStatus.Builder finishedStatusBuilder = CompactionTaskFinishedStatus.builder()
                         .finishTime(getInstantAttribute(item, FINISH_TIME))
                         .timeSpentOnJobs(Duration.ofMillis(getLongAttribute(item, MILLIS_SPENT_ON_JOBS, 0)))
                         .totalJobRuns(getIntAttribute(item, NUMBER_OF_JOBS, 0))
                         .totalRecordsRead(getLongAttribute(item, LINES_READ, 0))
-                        .totalRecordsWritten(getLongAttribute(item, LINES_WRITTEN, 0))
-                        .recordsReadPerSecond(Double.parseDouble(getNumberAttribute(item, READ_RATE)))
-                        .recordsWrittenPerSecond(Double.parseDouble(getNumberAttribute(item, WRITE_RATE)))
-                        .build());
+                        .totalRecordsWritten(getLongAttribute(item, LINES_WRITTEN, 0));
+                getDoubleAttribute(item, READ_RATE).ifPresent(finishedStatusBuilder::recordsReadPerSecond);
+                getDoubleAttribute(item, WRITE_RATE).ifPresent(finishedStatusBuilder::recordsWrittenPerSecond);
```

Comment:
It would be nice to keep this builder all in-line. With an optional you can do that with orElse.